web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -845,7 +849,79 @@ private static void warnLegacyTicketTypes(String ticketType) {
         }
     }
 
-    @JsMethod
+    public Promise<SharedExportBytesUnion> shareObject(ServerObject object, SharedExportBytesUnion sharedTicketBytes) {
+        if (object.getConnection() != this) {
+            return Promise.reject("Cannot share an object that comes from another server instance");
+        }
+        PublishRequest request = new PublishRequest();
+        request.setSourceId(object.typedTicket().getTicket());
+
+        Ticket ticket = sharedTicketFromStringOrBytes(sharedTicketBytes);
+        request.setResultId(ticket);
+
+        return Callbacks.grpcUnaryPromise(c -> {
+            sessionServiceClient().publishFromTicket(request, metadata(), c::apply);
+        }).then(ignore -> Promise.resolve(sharedTicketBytes));
+    }
+
+    private static Ticket sharedTicketFromStringOrBytes(SharedExportBytesUnion sharedTicketBytes) {
+        Ticket ticket = new Ticket();
+        final int length;
+        final TypedArray.SetArrayUnionType array;
+        if (sharedTicketBytes.isString()) {
+            byte[] arr = sharedTicketBytes.toString().getBytes(StandardCharsets.UTF_8);
+            length = arr.length;
+            array = TypedArray.SetArrayUnionType.of(arr);
+        } else {
+            Uint8Array bytes = (Uint8Array) sharedTicketBytes;
+            length = bytes.length;
+            array = TypedArray.SetArrayUnionType.of(bytes);
+        }
+        Uint8Array bytesWithPrefix = new Uint8Array(length + 2);
+        bytesWithPrefix.setAt(0, (double) 'h');
+        bytesWithPrefix.setAt(1, (double) '/');
```

Comment:
- What is this magic `h/`, what does it mean? Comment would be useful here.

Reply:
Every ticket type has a one char prefix followed by a slash - `s` is for scope tickets, `e` is for export tickets. The `h` prefix is what the server documents for shared tickets.

Reply:
Is there anywhere all the prefixes are documented?

Reply:
No - scope handling is technically optional aside from export tickets, downstream server implementations can skip various ticket resolvers, or add their own.

Reply:
This is non-optional - all tickets being sent to a DH server must have a prefix of this form. Shared tickets never must supply this (in fact there is a test explicitly to confirm that `h/foo` is _not_ the same as `foo`, just in case of this kind of confusion) unless it is part of their own naming scheme.

There's no magic here, this is always required for export/scope/application tickets, and for any other custom ticket resolver that any downstream implementation adds (see DHE's database tickets, for example). Application/scope tickets already come to the JS API consumer with the prefix added, and export tickets are created with this on the server (see ExportTicketHelper.exportIdToBytes()) and on the client (see ClientConfiguration.newTicketRaw()). This is very consistently applied and not at all magic.

Client code calling this should never care about h/ prefixes, and can make up their own naming schemes with total disregard for the implementation details.